test_python/test_c2.py

```diff
@@ -65,6 +63,42 @@ def ref(X, W):
             reference=ref,
         )
 
+    @given(n=st.integers(1, 128),
```

Comment:
can you add 1-2 lines for the logic behind how autotuner works atm? I think that the current solution is not stable and a better solution can be coded up easily/quickly.